docs/NewFlux_Comentary.md

```diff
@@ -0,0 +1,251 @@
+
+Similarly the `aggregate` and `select` functions will have parameters for `windowing`, that enable you to define how you would like to group the data in time.
+
+The `group` and `window` functions are not part of this new Flux vision.
+
+### Syntax Sugar for Aggregates
+
+Aggregates also come with their own new syntax. The intent of this syntax is to make it easy to describe which aggregate function to apply to which columns.
+This syntax allows for that definition to be explicit and terse.
+
+The syntax is:
+
+```
+ |> aggregate({min(degrees_bottom), max(degrees_bottom), mean(degrees_bottom)}, by:["location"])
+```
+
+The semantics of this syntax are to apply three different aggregate functions to the same column named `degrees_bottom`.
+The output table will then have three columns named `min_degrees_bottom`, `max_degrees_bottom`, and `mean_degrees_bottom`.
+
+
+>NOTE: Selectors do not need their own syntax as it is easy to specify selection in the existing syntax.
+
+### Column Indexing
+
+A consequence of data being represented as a single table the `tableFind` function is not needed. As such  a user can call `getColumn` directly on a table.
+We can go one step further and use the `.` syntax to mean indexing a table for its column.
+
+For example:
+
+```
+data = @foo.bar{location, state == "CA"}
+
+data.location // a list of locations in the state of CA
+```
+
+
+## The How
+
+Now that we have an overview of the new features syntax, let's discuss how I expect they can be implemented.
+Again this is just to seed the conversation and is by no means a prescription on how it must be done.
+
+
+### Single Table vs Stream of Tables
+
+The current Flux internals use the interpreter to create DAG of promises to data to construct a query spec.
+That spec is then planned etc and passed down to the execution engine.
+
+So long as we can continue to build a query spec from this new Flux API, we can continue to execute the queries.
+My rough plan is that a set of new Flux functions will be added to a `table` package.
+The `table` package will be very similar to the current `universe` package. It will contain all basic functions and new functions as well like `aggregate`.
+
+This new `table` package will need to provide its own mechanism for producing a query spec.
+The current execution engine can handle streams of tables with heterogenous schemas.
+A homogenous schema is a subset of functionality in a heterogeneous schema and as such can run on the execution engine without issue.
+We will still leverage the engines concept of groups to do aggregates etc but those details will be hidden from the user.
+
+As a rough example to illustrate how this would work, the following new Flux:
+
+
+```
+@data.temperatures{location,state,degrees_bottom}
+ |> aggregate({min(degrees_bottom)}, by:["location"])
+```
+
+would generate a spec similar to what this Flux would generate
+
+```
+from(bucket:"data")
+ |> filter(fn:(r) => r._measurement == "temperatures")
```

Comment:
There is a missing filter predicate `r._field == "degrees_bottom"`, right?  It seems like we would need to know which columns are InfluxDB tags and which are fields statically to make this transformation?

Reply:
I'm confused. Do we need to filter on the field here in order to perform an aggregate function on it?

Reply:
I guess since the `from` function will produce pivoted data, there will be no `_field` column.  But I still don't see how we specify which fields from a measurement to get from storage.  Is it all of then, by default?

Reply:
@wolffcm I was thinking we would send the entire list of columns down to storage and let it figure it out compared to the series index. It should be able to work out which columns are tags and which are fields at that point. Then we do the pivot directly as part of the `from` implementation for only the fields that are returned.

Does that sound like it would work?

Reply:
It seems doable.  I don't think we have any fields vs tags logic in the read path in storage currently, at least for Flux.  It seems like we'll need to change our interface with storage significantly to make this work correctly and efficiently.